src/main/java/uk/ac/ebi/eva/pipeline/io/readers/PedReader.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2016-2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.pipeline.io.readers;
+
+import org.opencb.biodata.formats.pedigree.io.PedigreePedReader;
+import org.opencb.biodata.formats.pedigree.io.PedigreeReader;
+import org.opencb.biodata.models.pedigree.Pedigree;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.batch.item.ItemReader;
+
+/**
+ * ItemReader that parses a PED file
+ * <p>
+ * PED specs
+ * http://pngu.mgh.harvard.edu/~purcell/plink/data.shtml#ped
+ */
+public class PedReader implements ItemReader<Pedigree> {
```

Comment:
This reader is making a resource acquisition, open and close operations besides the read operation. Please change the reader to an appropriate ResourceAwareItemReaderItemStream or any of it's implementations or extensions that fit better for this use case.